src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs

```diff
@@ -107,6 +116,110 @@ private void LoginButton_OnClick(object sender, RoutedEventArgs e)
                 }
             }
         }
+
+        public List<Control> AllChildren(DependencyObject parent)
+        {
+            var _list = new List<Control> { };
+            for (int i = 0; i < VisualTreeHelper.GetChildrenCount(parent); i++)
+            {
+                var _child = VisualTreeHelper.GetChild(parent, i);
+                if (_child is Control)
+                {
+                    _list.Add(_child as Control);
+                    _list.AddRange(AllChildren(_child));
+                }
+            }
+            return _list;
+        }
+
+        private Button GetButton(string buttonKey)
+        {
+            if (this.shortcutBarItems.Count > 1)
+            {
+                try
+                {
+                    int buttonIndex = ShortcutBarItems.ToList().FindIndex(item => item.ShortcutToolTip.ToUpper().Contains(buttonKey));
+                    var _container = ShortcutItemsControl.ItemContainerGenerator.ContainerFromIndex(buttonIndex);
+                    var _children = AllChildren(_container);
```

Comment:
Do we have to retrieve the button in this complex way? Can we grab it by name?

Reply:
actually We don't have access to the new, open and save buttons by name since they are dynamically created

Reply:
Their class `ShortcutBarItem` is our own class, we can modify the class to add properties for searching..

Reply:
I guess my point is that lets not rely on the tooltip to search for controls

Reply:
Done.